free/src/main/scala/cats/free/Free.scala

```diff
@@ -250,3 +250,40 @@ object Free {
       def flatMap[A, B](a: Free[S, A])(f: A => Free[S, B]): Free[S, B] = a.flatMap(f)
     }
 }
+
+private trait FreeFoldable[F[_]] extends Foldable[Free[F,?]] {
+
+  implicit def F: Foldable[F]
+  implicit def FunctorF: Functor[F]
+
+  override final def foldLeft[A,B](fa: Free[F,A], b: B)(f: (B,A) => B): B =
+    fa.fold(f(b,_), F.foldLeft(_,b)((bb,freeA) => foldLeft(freeA,bb)(f)))
+
+  override final def foldRight[A,B](fa: Free[F,A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+    fa.fold(f(_, lb), F.foldRight(_,lb)( (freeA,eb) => foldRight(freeA,eb)(f)))
+}
+
+private trait FreeTraverse[F[_]] extends Traverse[Free[F,?]] with FreeFoldable[F] {
```

Comment:
You can override `map` here to use `Free#map` which is more efficient than the default `map` for `Traverse` (`traverse` with `Id`).